vector_store.go

```diff
@@ -0,0 +1,349 @@
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, nil)
+	return
+}
+
+// ListVectorStoreFiles Lists the currently available files for a vector store.
+func (c *Client) ListVectorStoreFiles(
+	ctx context.Context,
+	vectorStoreID string,
+	limit *int,
+	order *string,
+	after *string,
+	before *string,
+) (response VectorStoreFilesList, err error) {
+	urlValues := url.Values{}
+	if after != nil {
+		urlValues.Add("after", *after)
+	}
+	if limit != nil {
+		urlValues.Add("limit", fmt.Sprintf("%d", *limit))
+	}
+	if before != nil {
+		urlValues.Add("before", *before)
+	}
+	if order != nil {
+		urlValues.Add("order", *order)
+	}
+
+	encodedValues := ""
+	if len(urlValues) > 0 {
+		encodedValues = "?" + urlValues.Encode()
+	}
+
+	urlSuffix := fmt.Sprintf("%s/%s%s%s", vectorStoresSuffix, vectorStoreID, vectorStoresFilesSuffix, encodedValues)
+	req, _ := c.newRequest(ctx, http.MethodGet, c.fullURL(urlSuffix),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// CreateVectorStoreFileBatch creates a new vector store file batch.
+func (c *Client) CreateVectorStoreFileBatch(
+	ctx context.Context,
+	vectorStoreID string,
+	request VectorStoreFileBatchRequest,
+) (response VectorStoreFileBatch, err error) {
+	urlSuffix := fmt.Sprintf("%s/%s%s", vectorStoresSuffix, vectorStoreID, vectorStoresFileBatchesSuffix)
+	req, _ := c.newRequest(ctx, http.MethodPost, c.fullURL(urlSuffix),
+		withBody(request),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// RetrieveVectorStoreFileBatch retrieves a vector store file batch.
+func (c *Client) RetrieveVectorStoreFileBatch(
+	ctx context.Context,
+	vectorStoreID string,
+	batchID string,
+) (response VectorStoreFileBatch, err error) {
+	urlSuffix := fmt.Sprintf("%s/%s%s/%s", vectorStoresSuffix, vectorStoreID, vectorStoresFileBatchesSuffix, batchID)
+	req, _ := c.newRequest(ctx, http.MethodGet, c.fullURL(urlSuffix),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// CancelVectorStoreFileBatch cancel a new vector store file batch.
+func (c *Client) CancelVectorStoreFileBatch(
+	ctx context.Context,
+	vectorStoreID string,
+	batchID string,
+) (response VectorStoreFileBatch, err error) {
+	urlSuffix := fmt.Sprintf("%s/%s%s/%s%s", vectorStoresSuffix,
+		vectorStoreID, vectorStoresFileBatchesSuffix, batchID, "/cancel")
+	req, _ := c.newRequest(ctx, http.MethodPost, c.fullURL(urlSuffix),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// ListVectorStoreFiles Lists the currently available files for a vector store.
+func (c *Client) ListVectorStoreFilesInBatch(
+	ctx context.Context,
+	vectorStoreID string,
+	batchID string,
+	limit *int,
+	order *string,
+	after *string,
+	before *string,
```

Comment:
Could we please use the [Pagination](https://github.com/sashabaranov/go-openai/blob/211cb49fc22766f4174fef15301c4d39aef609d3/run.go#L210) struct here?